src/ProductView/Diagram/LineChart.js

```diff
@@ -21,6 +25,27 @@ class LineChart extends Component {
         return (nextProps.events.length !== this.props.events.length
         || nextProps.eventType.name !== this.props.eventType.name);
     }
+
+    getDiagramColors() {
+        const state = this.props.product.state.toLowerCase();
+        switch(state) {
+            case argosConfig.runningStateName:
+                this.backgroundColor = argosConfig.runningStateTransparentColor;
+                this.borderColor = argosConfig.runningStateColor;
+                break;
+            case argosConfig.warningStateName:
+                this.backgroundColor = argosConfig.warningStateTransparentColor;
+                this.borderColor = argosConfig.warningStateColor;
+                break;
+            case argosConfig.errorStateName:
+                this.backgroundColor = argosConfig.errorStateTransparentColor;
+                this.borderColor = argosConfig.errorStateColor;
+                break;
+            default:
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Either merge this case with the identical one on line "32" or change one of the implementations. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=javascript%3AS1871)